### PR TITLE
Create a new deployment on every Jekyll build

### PR DESCRIPTION
**Pull request by nikitag-ms**
Example at https://github.com/Paper-SPA/nikitag-test/deployments:
image

### CODE REVIEW

File: deploy/gh-deployments.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+# Make the script more scrict (stop on errors including in pipes)
+set -e
+set -o pipefail
+
+# Create a deployment
+# https://developer.github.com/v3/repos/deployments/#create-a-deployment
+createGitHubDeployment() {
+    RESPONSE=$(curl --silent --show-error --fail \
```
Comment (YiMysty) [inline deploy/gh-deployments.sh]: 2 spaces vs 4 spaces

File: deploy/gh-deployments.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        -H "Content-Type: text/json; charset=utf-8" \
+        -H "Accept: application/vnd.github.flash-preview+json" \
+        -d @- <<EOF
+{
+  "environment": "production",
+  "state": "$status",
+  "description": "$description"
+}
+EOF
+}
```
Comment (YiMysty) [inline deploy/gh-deployments.sh]: extra line

File: deploy/run.sh [hunk 3]
```diff
@@ -56,4 +72,9 @@ cd "${PAGES_SITE_FOLDER}/" && zip -r "$ZIP_FILE_NAME" ./*
 STATUS_CODE=$(curl --write-out "%{http_code}" -o /dev/null -H "Authorization: token $ACCESSTOKEN" -F "app=@$ZIP_FILE_NAME" -F "event=@${GITHUB_EVENT_PATH}" "$PAGES_UPLOAD_HOST")
 if [ "$STATUS_CODE" -ne 200 ]; then
   echo "::error ::Deploy to the app service failed with status code $STATUS_CODE" 
+  # Update GitHub deployment status
+  postGitHubDeploymentStatus "${DEPLOYMENT_ID}" "failure" "Deploy to the app service failed. See Actions output for details."
 fi
+
+# Update GitHub deployment status
+postSuccessGitHubDeploymentStatus "${DEPLOYMENT_ID}" "${PAGES_ACCESS_URL}"
```
Comment (YiMysty) [inline deploy/run.sh]: extra line in the end

File: deploy/run.sh [hunk 4]
```diff
@@ -7,17 +7,24 @@ set -o pipefail
...
+
+# Create a GitHub deployment (default status -- Running)
+DEPLOYMENT_ID=$(createGitHubDeployment)
 
 # Provision the service
 RESPONSE=$(curl -X POST "$PAGES_PROVISIONING_HOST/v0.1/spa/provision?repository=${GITHUB_REPOSITORY}" -H "Content-Length: 0" -H "GitHub-Token: ${GITHUB_TOKEN}")
 OPERATIONID=$(echo "$RESPONSE" | jq -r .operationId)
 if [ -z "$OPERATIONID" ]; then
   echo "::error ::Failed to send request to provisioning service with response $RESPONSE"
+  # Update GitHub deployment status
```
Comment (YiMysty) [inline deploy/run.sh]: one empty line before each comment

File: deploy/gh-deployments.sh [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+"required_contexts": []
+}
+EOF
+)
+    echo "$RESPONSE" | jq -r .id
+} 
+
+# Post successfull deployment status
+# https://developer.github.com/v3/repos/deployments/#create-a-deployment-status
+postSuccessGitHubDeploymentStatus() {
```
Comment (YiMysty) [inline deploy/gh-deployments.sh]: can you combine `postSuccessGitHubDeploymentStatus` and `postGitHubDeploymentStatus` into one function? :) most of code are identical.

**Comment by YiMysty** [comment]
script need lint https://github.com/Paper-SPA/pages-actions/pull/11/checks?check_run_id=309150671

File: deploy/run.sh [hunk 6]
```diff
@@ -7,17 +7,24 @@ set -o pipefail
 # Load shared functions
 # shellcheck source=deploy/functions.sh
```
Comment (yoannchaudet) [inline deploy/run.sh]: You will need to duplicate this comment which is a special instruction for shellcheck (or bash "linter") to find your file when it runs.

File: deploy/gh-deployments.sh [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+createGitHubDeployment() {
+    RESPONSE=$(curl --silent --show-error --fail \
+        -X POST "${GITHUB_API_URL}/repos/${GITHUB_REPOSITORY}/deployments" \
+        -H "Authorization: token ${GITHUB_TOKEN}" \
+        -H "Content-Type: text/json; charset=utf-8" \
+        -H "Accept: application/vnd.github.flash-preview+json" \
+        -d @- <<EOF
+{
+"ref": "$GITHUB_SHA",
+"environment": "production",
```
Comment (yoannchaudet) [inline deploy/gh-deployments.sh]: Shall this be a parameter too? It is repeated in multiple places.

Comment (nikitag-ms) [reply deploy/gh-deployments.sh]: Environment? I'll just make it a variable in the file. I don't think we are going to pass a value different from `production`

File: deploy/gh-deployments.sh (hunk 5 above)
Comment (nikitag-ms) [reply deploy/gh-deployments.sh]: Can we do this as part of a separate WI? My bash skills are failing me. In addition, it looks like Actions are down (at the time of writing), so I can't even test it anymore.

Comment (nikitag-ms) [reply deploy/gh-deployments.sh]: Spoke offline, we decided to go ahead with the merge. Thank you!

**Pull request merged**